core/src/main/java/jenkins/model/JenkinsLocationConfiguration.java

```diff
@@ -211,10 +210,40 @@ public FormValidation doCheckUrl(@QueryParameter String value) {
 
     public FormValidation doCheckAdminAddress(@QueryParameter String value) {
         try {
-            new InternetAddress(value);
+            Class<?> internetAddress = Jenkins.get().pluginManager.uberClassLoader.loadClass("javax.mail.internet.InternetAddress");
```

Comment:
could we just replace this whole method with something like:

```java
Pattern.compile("^(.+)@(\\S+)$")
      .matcher(value)
      .matches()
```

(obviously can make the pattern a constant)

from https://www.baeldung.com/java-email-validation-regex

Reply:
Possibly. Feel free to propose a separate PR doing so if desired. This PR is maintaining the existing status quo.

Reply:
=/ I don't think that adding 30 lines of reflection is 'maintaining' the status quo.

If we're removing a dependency then we either should have a simple way of validating it or removing the validation.

Reply:
It _is_ maintaining the status quo in two senses:

- Before, we called `InternetAddress(String)` to do the validation. This PR continues to call the same validation routine.
- Before, when detaching functionality we have used reflection to maintain anti-modular interfaces (e.g., `AbstractBuild#getTestResultAction` and `AbstractBuild#getAggregatedTestResultAction`. This PR follows the same precedent.

I am not going to change the status quo in this PR. If you feel that this implementation should be replaced with a regular expression or deleted, feel free to propose a separate PR for that. If and when it is accepted, I will merge it with this PR (if this PR isn't shipped first).

Reply:
> when detaching functionality we have used reflection to maintain anti-modular interfaces (e.g., `AbstractBuild#getTestResultAction` and `AbstractBuild#getAggregatedTestResultAction`)

Those cases were unfortunate but required to prevent serious regression of basic CI functionality; there was just no way around it. This bit of form validation (on what I am guessing is a rather unimportant setting—in core, seems to be used only for RSS feeds) does not in my opinion justify use of reflection.

Reply:
In my opinion it is good enough, and it is all I am willing to offer. Take it or leave it. If you don't want this PR, feel free to close it.

Reply:
> feel free to propose a separate PR for that

in progress

Reply:
Rebased on top of #6166